scripts/dotnet.py

```diff
@@ -632,10 +633,16 @@ def get_commit_date(
                 success = 1
         except URLError:
             retrycount += 1
+            getLogger().warning(f"URL Error trying to get commit date from {url}, Attempt {retrycount}")
+            sleep(60)
 
     if not build_timestamp:
-        raise RuntimeError(
-            'Could not get timestamp for commit %s' % commit_sha)
+        if not item:
+            raise RuntimeError(
+                'Could not get timestamp for commit %s' % commit_sha)
+        else:
+            raise RuntimeError(
+                'Could not get timestamp for commit %s, data retrieved:\n %s' % (commit_sha, item))
```

Comment:
Is this a real scenario we have seen? Trying to understand why the API would return successfully but not contain the build timestamp

Reply:
Since success is not the only breakout factor currently, we may not have a timestamp if we run out of retries and quit after failing.

Reply:
You are right, I had been still figuring out where the exception might be thrown and it is thrown before item gets assigned. I don't think we have seen the case with success but no datetime, although it is hard to know for sure.